### PR TITLE
Host sprite manifest on s3

### PR DESCRIPTION
**Pull request by bethanyaconnor**
We want to localize the ability to search the animation library in sprite lab. In order to do that, we'll need localized copies of the sprite manifest, which is fairly large (too large to reasonably store ~80 copies). Instead, we should hoist the manifest up to S3 and fetch it on page load. Currently, I'm only implementing this for spritelab and gamelab will still fetch a local JSON file.

Right now, we don't have localized versions of the manifest, so we are always fetching the same file. In the future, we'll fetch a different file dependent on the user's language.

I haven't uploaded the manifest to S3 yet in case there's any feedback on where it should go in the bucket. I'll do that before merging this PR.

## Links

- [tech spec](https://docs.google.com/document/d/1ri9Qbw6KVXkKK1DibL_bIwKwV_cnjdzwbUKgHX9_Dt8/edit)
- [jira](https://codedotorg.atlassian.net/browse/FND-1142)

## Testing story

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/p5lab/AnimationPicker/AnimationPickerBody.jsx [hunk 1]
```diff
@@ -44,31 +44,16 @@ export default class AnimationPickerBody extends React.Component {
     onPickLibraryAnimation: PropTypes.func.isRequired,
     onUploadClick: PropTypes.func.isRequired,
     playAnimations: PropTypes.bool.isRequired,
-    libraryManifest: PropTypes.object.isRequired,
+    getLibraryManifest: PropTypes.func.isRequired,
     categories: PropTypes.object.isRequired,
     hideUploadOption: PropTypes.bool.isRequired
   };
 
-  constructor(props) {
```
Comment (bethanyaconnor) [inline apps/src/p5lab/AnimationPicker/AnimationPickerBody.jsx:52]: We don't know for sure that the manifest has loaded here. We could either change the spritelab code to be async and wait, or be greedy and only call `searchAssets` when we need to search. I tested removing this (plus the changes below) and it worked fine for me but let me know if I missed a case here.

File: apps/src/p5lab/P5LabView.jsx [hunk 2]
```diff
@@ -102,7 +110,7 @@ class P5LabView extends React.Component {
             <AnimationPicker
               channelId={this.getChannelId()}
               allowedExtensions=".png,.jpg,.jpeg"
-              libraryManifest={this.getLibraryManifest()}
+              getLibraryManifest={() => this.getLibraryManifest()}
```
Comment (maddiedierker) [inline apps/src/p5lab/P5LabView.jsx]: if you refactor `this.getLibraryManifest` into an arrow function, this line can simplify to:
```
getLibraryManifest={this.getLibraryManifest}
```

Comment (maddiedierker) [reply apps/src/p5lab/P5LabView.jsx]: (same for the case below on line 141)

File: apps/src/p5lab/AnimationPicker/AnimationPickerBody.jsx (hunk 1 above)
Comment (maddiedierker) [reply apps/src/p5lab/AnimationPicker/AnimationPickerBody.jsx:52]: very interesting! i may have unnecessarily refactored this code into what it is now. the expected/wanted behavior when this component loads is that it shows the "draw your own" option along with all of the available categories -- is that what currently happens?

**Comment by maddiedierker** [comment]
looping @ajpal in as well because she has a task this sprint to refactor the aliases in the manifest to be space-delimited ([STAR-1133](https://codedotorg.atlassian.net/browse/STAR-1133)) and this PR should probably be merged before she starts on that

File: apps/src/p5lab/AnimationPicker/AnimationPickerBody.jsx (hunk 1 above)
Comment (bethanyaconnor) [reply apps/src/p5lab/AnimationPicker/AnimationPickerBody.jsx:52]: I think this looks right?

Spritelab:
antimation-tab

Gamelab:
gamelab-animation-tab

Comment (maddiedierker) [reply apps/src/p5lab/AnimationPicker/AnimationPickerBody.jsx:52]: it does! yay for deleting code 🎉 i wish i had noticed that when i refactored this constructor 😂

**Review by maddiedierker — APPROVED** [review]
👯

**Comment by bethanyaconnor** [comment]
Manifest uploaded to S3 -- things look good on production (as expected). Things also look good locally using the manifest off S3. I'm going to let this PR go through Drone (though I don't expect it would catch anything as it didn't fail before) then merge.

**Pull request merged**